frameworklauncher/src/main/java/com/microsoft/frameworklauncher/common/model/SummarizedFrameworkInfos.java

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved. 
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING 
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND 
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, 
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, 
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE. 
+
+package com.microsoft.frameworklauncher.common.model;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SummarizedFrameworkInfos implements Serializable {
+  private List<String> summarizedFrameworkInfos = new ArrayList<>();
+
+  public void add(FrameworkRequest frameworkRequest, FrameworkStatus frameworkStatus) {
+    FrameworkDescriptor frameworkDescriptor = frameworkRequest.getFrameworkDescriptor();
+
+    // Notes:
+    // 1. To ensure the result string can be converted back by spiting by comma,
+    //    all items should not contain comma.
+    // 2. To ensure no breaking changes, new item should be added to the tail .
+    String summarizedFrameworkInfo = frameworkRequest.getFrameworkName() + "," +
+        frameworkDescriptor.getVersion() + "," +
+        frameworkDescriptor.getDescription() + "," +
+        frameworkDescriptor.getUser().getName() + "," +
+        frameworkRequest.getFirstRequestTimestamp() + "," +
+        frameworkRequest.getLastRequestTimestamp() + "," +
+        frameworkStatus.getFrameworkState() + "," +
+        frameworkStatus.getFrameworkRetryPolicyState().computeAllRetriedCount() + "," +
+        frameworkStatus.getFrameworkCompletedTimestamp() + "," +
+        frameworkStatus.getApplicationExitCode();
+
```

Comment:
Flatten framework info into strings not friendly for callers such as REST API. The original JSON notation for each framework info is self explainable and better than this.

Reply:
For 2500 frameworks, but original JSON notation increase the payload from 208K to 1.4MB, the round trip time will increase from 100ms to ~700ms.
So, it maybe worth to remove notations.

---
In reply to: [166541056](https://github.com/Microsoft/pai/pull/309#discussion_r166541056) [](ancestors = 166541056)

Reply:
Revert as your suggested, tks!

---
In reply to: [166541632](https://github.com/Microsoft/pai/pull/309#discussion_r166541632) [](ancestors = 166541632,166541056)